src/Sentry/Platforms/Cocoa/SentrySdk.cs

```diff
@@ -86,29 +87,111 @@ private static void InitSentryCocoaSdk(SentryOptions options)
-        //     };
-        // }
+        nativeOptions.BeforeSend = evt =>
+        {
+            // When we have an unhandled managed exception, we send that to Sentry twice - once managed and once native.
+            // The managed exception is what a .NET developer would expect, and it is sent by the Sentry.NET SDK
+            // But we also get a native SIGABRT since it crashed the application, which is sent by the Sentry Cocoa SDK.
+
+            // There should only be one exception on the event in this case
+            if (options.Native.SuppressSegfaults && evt.Exceptions?.Length == 1)
+            {
+                // It will match the following characteristics
+                var ex = evt.Exceptions[0];
+
+                // Thankfully, sometimes we can see Xamarin's unhandled exception handler on the stack trace, so we can filter
+                // them out. Here is the function that calls abort(), which we will use as a filter:
+                // https://github.com/xamarin/xamarin-macios/blob/c55fbdfef95028ba03d0f7a35aebca03bd76f852/runtime/runtime.m#L1114-L1122
+                if (ex.Type == "SIGABRT" && ex.Value == "Signal 6, Code 0" &&
+                    ex.Stacktrace?.Frames.Any(f => f.Function == "xamarin_unhandled_exception_handler") is true)
+                {
+                    // Don't send it
+                    options.LogDebug("Discarded {0} error ({1}). Captured as  managed exception instead.", ex.Type,
+                        ex.Value);
+                    return null!;
+                }
+
+                // Similar workaround for NullReferenceExceptions. We don't have any easy way to know whether the
+                // exception is managed code (compiled to native) or original native code though.
+                // See: https://github.com/getsentry/sentry-dotnet/issues/3776
+                if (ex.Type == "EXC_BAD_ACCESS")
+                {
+                    // Don't send it
+                    options.LogDebug("Discarded {0} error ({1}). Captured as  managed exception instead.", ex.Type,
+                        ex.Value);
+                    return null!;
+                }
+            }
+
+            // we run our SIGABRT checks first before handing over to user events
+            // because we delegate to user code, we need to protect anything that could happen in this event
+            if (options.BeforeSendInternal == null)
+                return evt;
+
+            try
+            {
+                var sentryEvent = evt.ToSentryEvent();
+                if (sentryEvent == null)
+                    return evt;
+
+                var result = options.BeforeSendInternal(sentryEvent, null!);
+                // returning null is fine - the native binding even has this set, but the tooling doesn't want to obey, so the pragma was necessary
+                if (result == null)
+                    return null!;
+
+                evt.ServerName = result.ServerName;
```

Comment:
I think the code on the following lines updates the subset of properties that might be modified by the `OnBeforeSend` callback right? Maybe worth adding that context as a comment (I'm only aware of it because I listened in on the chat with @bruno-garcia).

We do try to let the code speak for itself and have/require as few comments as possible. In this case, I think it's some non-obvious requirements we landed on after a bit of back and forth though.